### ISSUE TITLE
PersistentVolume erases /opt/app-root/src

### ISSUE THREAD

**Issue by akarasev** [sclorg/s2i-php-container#222]
Steps to reproduce:

1. Deploy application `centos/php-70-centos7`
2. Go to OpenShift Origin Console > Applications > Deployments > (choose application) > Configuration tab > Volumes Add Storage > Add

Expected behavior: application still works

Actual behavior: application doesn't work because `/opt/app-root/src` is empty now.

**Comment by omron93** [comment]
@bparees Could you please look into this?

**Comment by bparees** [comment]
what mount point are you setting when you add the storage?

**Comment by akarasev** [comment]
@omron93 @bparees thanks for your answer. The only thing I confuse is why it works on Docker. Let me explain:

```
$ s2i build https://github.com/salesagility/SuiteCRM.git centos/php-70-centos7 php-test-app
...
$ docker volume ls
DRIVER              VOLUME NAME
$ docker run -p 8080:8080 -v app-data:/opt/app-root/src --name php-app php-test-app
582a6be3d86203a8d4ecf5f5c5b2b76db500839c9742a4e115343662222d4e24
$ docker restart php-app 
php-app
$ curl --head localhost:8080
HTTP/1.1 302 Found
Date: Fri, 03 Aug 2018 05:12:11 GMT
Server: Apache/2.4.27 (Red Hat) OpenSSL/1.0.2k-fips
Location: install.php
Content-Type: text/html; charset=UTF-8
```

Could you please explain me what's the difference between pure Docker and OpenShift Origin?
What is a recommended way to save the content `opt/app-root/src` on OpenShift Origin's PersistentVolume?

**Comment by bparees** [comment]
@akarasev in your scenario you have effectively asked docker to create a new volume and mount it into your container.  When you do that, docker will indeed populate the new volume with the content from the image.

If instead you had mounted an existing dir/volume into your container (which is what openshift is effectively doing on your behalf here), you would see that the image content is overwritten.

To more closely simulate what openshift is doing, you would have to do the following:

$ s2i build https://github.com/salesagility/SuiteCRM.git centos/php-70-centos7 php-test-app
$ mkdir /tmp/myvolume
$ docker run -p 8080:8080 -v /tmp/myvolume:/opt/app-root/src --name php-app php-test-app

At which point you'll see that /opt/app-root/src inside the container is empty.

**Comment by akarasev** [comment]
@bparees are you saying Docker's named volumes != OpenShift PersistentVolume?
Sounds like a common pitfall which might be described in documentation for this repository.
Could you give some advice how to force OpenShift mount not a dir, but Docker named volume analog?

My goal here to have persistent `/opt/app-root/src` content.

**Comment by bparees** [comment]
@akarasev no, i'm not saying that.  i'm saying that when you use an openshift volume, you're using the equivalent of a pre-existing directory with docker.

I showed you how, when using docker, you can see the exact same behavior as what you are seeing in openshift.

> My goal here is to have persistent /opt/app-root/src content.

The only way you're going to get there is to mount the volume somewhere and have your `run` script for the image copy the content you want to the mounted location, and then run the application from that location.

**Comment by akarasev** [comment]
@bparees, it's clear for me, now. Thanks for your help!
